src/ecs/resources/input.rs

```diff
@@ -62,45 +92,63 @@ impl<'a> Iterator for PressedMouseButtons<'a> {
 /// ```
 #[derive(Default)]
 pub struct InputHandler {
-    pressed_keys: HashMap<VirtualKeyCode, KeyQueryState>,
-    pressed_mouse_buttons: HashMap<MouseButton, KeyQueryState>,
+    pressed_keys: Vec<VirtualKeyCode>,
+    previous_pressed_keys: Vec<VirtualKeyCode>,
+    pressed_mouse_buttons: Vec<MouseButton>,
+    previous_pressed_mouse_buttons: Vec<MouseButton>,
     // defined to match glium
     mouse_position: Option<(i32, i32)>,
     previous_mouse_position: Option<(i32, i32)>,
+    axes: HashMap<i32, Axis>,
```

Comment:
I assume these `i32` are specific indices of the elements, in which case it would be a good idea to make them opaque (i.e. `struct AxeId(i32)`) so that the user is not able to change them